[OpenMP] Directly use user's grid and block size in kernel language mode

shiltian opened the pull request:
In kernel language mode, use user's grid and blocks size directly. No validity
check, which means if user's values are too large, the launch will fail, similar
to what CUDA and HIP are doing right now.

shiltian commented:
Gentle ping

jdoerfert commented on clang/lib/Sema/SemaOpenMP.cpp:
```diff
@@ -14633,6 +14633,26 @@ StmtResult Sema::ActOnOpenMPTargetTeamsDirective(ArrayRef<OMPClause *> Clauses,
+  bool HasBareClause = false;
+  bool HasThreadLimitClause = false;
+  bool HasNumTeamsClause = false;
+  OMPClause *BareClause = nullptr;
```
No need for HasBareClause since you got the pointer.

jdoerfert reviewed (approved):
LG, see the nit.

Also, add a runtime test with bare and verify we stick with values we otherwise would not, e.g., 10000 teams and 1024 threads.

shiltian committed:
[OpenMP] Directly use user's grid and block size in kernel language mode

In kernel language mode, use user's grid and blocks size directly. No validity
check, which means if user's values are too large, the launch will fail, similar
to what CUDA and HIP are doing right now.

shiltian commented:
The format of the test will be ignored.
